### DISCUSSION TITLE
fastq-dump ... ERROR

### DISCUSSION THREAD

**Discussion by PC2021** [biostars/biostar-handbook-issues#251] [General]
Hi there! I have been following the Step-by-Step installation and got two errors as you can see below. Could anyone help me?

$ ~/bin/doctor.py
# Doctor! Doctor! Give me the news.
# Checking symptoms ...
# bwa ... OK
# datamash ... OK
# fastqc -h ... OK
# hisat2 ... OK
# seqret -h ... OK
# featureCounts ... OK
# efetch ... OK
# esearch ... OK
# samtools ... OK
# fastq-dump ... ERROR
# bowtie2 ... OK
# bcftools ... OK
# seqtk ... OK
# seqkit ... OK
# bio ... OK
# **fastq-dump -X 1 -Z SRR1553591 ... ERROR**
#
# Your system shows 2 errors!

I tried to fix, and I got the following error message.

$ fastq-dump --stdout -X 2 SRR390728
Can't locate XML/LibXML.pm in @INC (you may need to install the XML::LibXML module) (@INC contains: /home/pdc/miniconda3/envs/bioinfo/lib/site_perl/5.26.2/x86_64-linux-thread-multi /home/pdc/miniconda3/envs/bioinfo/lib/site_perl/5.26.2 /home/pdc/miniconda3/envs/bioinfo/lib/5.26.2/x86_64-linux-thread-multi /home/pdc/miniconda3/envs/bioinfo/lib/5.26.2 .) at /home/pdc/miniconda3/envs/bioinfo/bin/fastq-dump line 13.
BEGIN failed--compilation aborted at /home/pdc/miniconda3/envs/bioinfo/bin/fastq-dump line 13.

**Comment by ialbert** [comment]
the step by step instruction has an alternative sratools install, 

follow that

**Comment by PC2021** [comment]
Thanks for your prompt reply. 

I followed "The step-by-step instructions too" and I got the same message errors.

I installed the XML::LibXML module again and I got the same error message.

Do you have any further instructions to solve this problem?

**Comment by ialbert** [comment]
no something is not right, you should not be getting any error of such sort,

fastq-dump is not a Perl script, you should not be getting this error at all,  there is no reason to start installing Perl modules

check where is fastq-dump running from

    which fastq-dump

what does it print?

**Comment by ialbert** [comment]
I was explicitly referring to look for the fastq-dump section not the rest of it

have yo installed the latest tools as described there?  have you restarted the terminal because that tool does not require perl,

**Reply by PC2021** [reply]
I restarted the terminal when I was installing the tools.

I ran the command "which fastq-dump" and I got the following path:

/home/pdc/miniconda3/envs/bioinfo/bin/fastq-dump

**Comment by ialbert** [comment]
yes you seem to still have the wrong fastqdum

have you done the step as indicated in the **How to fix fastq-dump errors**

      curl http://data.biostarhandbook.com/sratools.sh | bash

you don't have to do the other steps, just that one.

**Reply by PC2021** [reply]
pdc@PDC ~
$ curl http://data.biostarhandbook.com/sratools.sh | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   898  100   898    0     0   2217      0 --:--:-- --:--:-- --:--:--  2217
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 81.8M  100 81.8M    0     0  12.2M      0  0:00:06  0:00:06 --:--:-- 17.7M
#
# Installation complete 
#
# Close and reopen the terminal.
#
# In the new terminal activate the bioinfo environment.
#
pdc@PDC ~
$ grep BIOSTAR_HANDBOOK_START ~/.bashrc
# >>> BIOSTAR_HANDBOOK_START
# >>> BIOSTAR_HANDBOOK_START

**Comment by PC2021** [comment]
Hi again!

I uninstalled and installed it three times. I tried installing it in a new directory but got the same error message. There seems to be a problem on line 13 as you can see in the error message: "...BEGIN failed--compilation aborted at /home/pdc/miniconda3/envs/bioinfo/bin/fastq-dump line 13..."

**Comment by PC2021** [comment]
Hi again!

I uninstalled and installed it three times. I tried installing it in a new directory but got the same error message. There seems to be a problem on line 13 as you can see in the error message: "...BEGIN failed--compilation aborted at /home/pdc/miniconda3/envs/bioinfo/bin/fastq-dump line 13

**Comment by ialbert** [comment]
I don't understand your problem, something else is going badly

fastq-dump is not a Perl script, there is no reason for Perl to run at all!

the second point is that the other fastq-dump should be found first, it should be located in 

    ~/bin/sratoolkit.3.0.0-mac64/bin/fastq-dump

on a mac, or

    ~/bin/sratoolkit.3.0.0-ubuntu64/bin/fastq-dump

on a Linux machine. And that path should be already activated.

Try running the command above.

Long story short, something else is wrong

**Reply by PC2021** [reply]
I ran the command below

$ ~/bin/sratoolkit.3.0.0-ubuntu64/bin/fastq-dump

This sra toolkit installation has not been configured.
Before continuing, please run: vdb-config --interactive
For more information, see https://www.ncbi.nlm.nih.gov/sra/docs/sra-cloud/

**Reply by ialbert** [reply]
what system are you running this on? the messages that you are getting are very a typical from what people experience

**Reply by ialbert** [reply]
just do what it says then, normally you would never need to that any of that

https://www.biostars.org/p/424404/

**Comment by PC2021** [comment]
I have been using:

$ cat /etc/os-release
NAME="Linux Mint"
VERSION="20.3 (Una)"
ID=linuxmint
ID_LIKE=ubuntu
PRETTY_NAME="Linux Mint 20.3"
VERSION_ID="20.3"
HOME_URL="https://www.linuxmint.com/"
SUPPORT_URL="https://forums.linuxmint.com/"
BUG_REPORT_URL="http://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/"
PRIVACY_POLICY_URL="https://www.linuxmint.com/"
VERSION_CODENAME=una
UBUNTU_CODENAME=focal

**Reply by PC2021** [reply]
I used this video [([(https://www.youtube.com/watch?v=gs4k3FHkQd0))](https://www.youtube.com/watch?v=gs4k3FHkQd0) to try to remove the error message reported above, but it didn't work.

**Reply by ialbert** [reply]
try this:

https://blogs.iu.edu/ncgas/2021/02/22/a-beginners-guide-to-the-sra/

in general read up on what it does and understand that, looks like your Linux is not recognized properly, the problem you have are really simple configuration issues, like a path not being present,

**Reply by ialbert** [reply]
(you don't need the module load steps, though)

**Reply by PC2021** [reply]
Hi, Ialbert! I haven't been able to fix the error yet. I'm using python 3.9.10 and not python 3.8. Do you think it may be generating a compatibility problem with the script? Best

**Reply by ialbert** [reply]
well, AFAIK most bioconda was built with 3.8 - you should use the commands that I published

**Reply by PC2021** [reply]
Hi, again!
I have tried to find the problem during the installation progress, and I got this message below. Do you think the problem I have been facing during the installation is due to this corrupted file?

# Installing the bio package
#
#
# Installing the Entrez Direct
#

gzip: rchive.Linux.gz: invalid compressed data--format violated
chmod: cannot access 'rchive.Linux': No such file or directory
#
#
# Biostar Handbook installation complete!

**Reply by ialbert** [reply]
the Entrez Direct installation step runs the commands that are described here:

https://www.ncbi.nlm.nih.gov/books/NBK179288/

rerun them separately, perhaps - though evidently there are some other things not quite optimal with your system

**Reply by PC2021** [reply]
Should I activate conda and write these commands?

**Reply by ialbert** [reply]
these should work independently of bioconda

**Reply by PC2021** [reply]
Hi, Ialbert! 
Good news! Everything is running appropriately.
The error message is coming up due to incompatibility between the python version and sratoolkit. Even using your command, bioconda was not updating to the latest version of Sratoolkit=3.0.0. I had to remove the sratoolkit=2.9.1.1 version and separately install sratoolkit=3.0.0. Additionally, I had to install one of the older versions of python, python=3.8.13. I have created other environments inside conda using the latest version of python and the newest version of sratoolkit. Also, it worked perfectly. I'm writing because this information may help other people with the same problem. 

Thanks for trying to help.

Best

**Reply by ialbert** [reply]
thanks for following up, bioconda is tested only with Python 3.8 thus that explains it
